### PR TITLE
Use Twemoji for emojis

### PR DESCRIPTION
**Pull request by mdwalters**
Not really sure if I should make a PR for this, but I can switch it to Twemoji if requested.

### CODE REVIEW

**Review by showierdata9978 — CHANGES_REQUESTED** [review]
never use @html, and install via npm

**Comment by showierdata9978** [comment]
sorry for the ping

**Comment by CST1229** [comment]
i think we should use twemoji

**Comment by mdwalters** [comment]
@CST1229 @showierdata9978 i fixed, and it works with all emojis now

**Comment by showierdata9978** [comment]
YOU STILL USE `@html` NEVER USE IT FOR USER GENERATED CONTENT

**Comment by CST1229** [comment]
you should probably escape the HTML (converting < into `@lt;` etc) instead of just sanitizing it

**Comment by mdwalters** [comment]
> YOU STILL USE `@html` NEVER USE IT FOR USER GENERATED CONTENT

and how am i going to get twemoji to work without `@html`?

**Comment by MikeDev101** [comment]
> One who does not learn from the past is doomed to repeat it.
https://www.youtube.com/watch?v=ykoACojHgeo

**Comment by CST1229** [comment]
> and how am i going to get twemoji to work without `@html`?

https://svelte.dev/docs#template-syntax-svelte-element

(probably parsing emojis manually instead of using a library)

**Comment by mdwalters** [comment]
> > One who does not learn from the past is doomed to repeat it.
> > https://www.youtube.com/watch?v=ykoACojHgeo

ok

**Review by showierdata9978 — DISMISSED** [review]
im not accepting until no `@html`

**Comment by mdwalters** [comment]
~~strict~~

**Comment by Bloctans** [comment]
i already did this just for it to be removed due to maxCDN being a tracking company

**Comment by mdwalters** [comment]
maxcdn shut down ._.

**Comment by showierdata9978** [comment]
actually the only way im going to allow Emojis is if we parse it ourselves. 

Twemoji uses innerHTML

**Comment by Bloctans** [comment]
should this be closed?

**Comment by showierdata9978** [comment]
needs to be fully redone

**Comment by mdwalters** [comment]
aaaaaaaaaaaaaaaaaaaaaaaaaaaaa

**Comment by showierdata9978** [comment]
still uses `@html`

File: src/lib/twemoji-utils.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+			text_split[i] = "<br>";
+		}
+	}
+	
+    const span = document.createElement("span");
+    const p = document.createElement("p");
+    p.innerText = text_split.join("");
+    span.appendChild(p);
+    return span;
+}
```
Comment (CST1229) [inline src/lib/twemoji-utils.js:19]: probably a better way to do this:
```suggestion
export function toHTMLElement(text) {
    const filteredText = text
        .replaceAll("&", "&amp;")
        .replaceAll("<", "&lt;")
        .replaceAll(">", "&gt;")
        .replaceAll('"', "&quot;")
        .replaceAll("'", "&apos;");
    
    const span = document.createElement("span");
    const p = document.createElement("p");
    p.innerText = filteredText;
    span.appendChild(p);
    return span;
}
```
and add `white-space: pre-wrap;` as a style to `.post-content` (so newlines and duplicated spaces are preserved)

**Comment by CST1229** [comment]
> still uses `@html`

that doesn't mean it's bad or vulnerable to XSS; you just have to be careful with it

**Comment by mdwalters** [comment]
> still uses `@html`

but it uses cst's escape trick

**Comment by mdwalters** [comment]
@showierdata9978 pls?

**Review by CST1229 — APPROVED** [review]

**Comment by tnix100** [comment]
@showierdata9978 please re-review.

**Comment by showierdata9978** [comment]
image

**Comment by CST1229** [comment]
> image

it's probably fine imo

**Comment by tnix100** [comment]
As long as it's not vulnerable to XSS, then it's fine.

**Review by tnix100 — APPROVED** [review]

**Review by CST1229 — APPROVED** [review]

**Comment by mdwalters** [comment]
> > 
> 
> Now we just gotta wait for showier to change their review.

*infinity years later...*

**Pull request merged**